utilities/message_filters/src/message_filters/__init__.py

```diff
@@ -121,19 +122,21 @@ def add(self, msg):
     def getInterval(self, from_stamp, to_stamp):
         """Query the current cache content between from_stamp to to_stamp."""
         assert from_stamp <= to_stamp
-        return [m for m in self.cache_msgs
-                if m.header.stamp >= from_stamp and m.header.stamp <= to_stamp]
+        return [self.cache_msgs[i] for i in range(0, len(self.cache_msgs))
+                if self.cache_times[i] >= from_stamp and self.cache_times[i] <= to_stamp]
```

Comment:
Please keep the list comprehension as simple as possible, e.g.:

    [ m for i, m in enumerate(self.cache_msgs) if self.cache_times[i] >= from_stamp and self.cache_times[i] <= to_stamp]